src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,47 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T, S> {
+    private T first;
+    private S second;
 
+    private Pair(T first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    static <T, S> Pair of(T t, S s) {
+        return new Pair(t, s);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        Pair<T, S> pair = (Pair<T, S>) obj;
+
+        if (first != null ? !first.equals(pair.first) : pair.first != null) {
```

Comment:
try to check fields without ternary operators
Hint: use only logic && and ||

Reply:
Done